### PR TITLE
Fix LifecycleListener's dispose method not being called on LWJGL3 backend

### PR DESCRIPTION
**Pull request by intrigus**
See https://github.com/libgdx/libgdx/issues/3673#issuecomment-172115381 and https://github.com/libgdx/libgdx/issues/3673#issuecomment-207808454 .

### CODE REVIEW

File: backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Application.java [hunk 1]
```diff
@@ -142,13 +143,31 @@ private void loop() {
...
+				}
+				closedWindow.dispose();
+
 				windows.removeValue(closedWindow, false);
 			}
 		}
 	}
 
 	private void cleanup() {
+		//synchronized (lifecycleListeners) {
```
Comment (intrigus) [inline backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Application.java]: I was not sure whether I should `synchronize` it or not.

Comment (badlogic) [inline backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Application.java]: Yes, please synchronize.

**Comment by badlogic** [comment]
Looks good. Please synchronize on the listeners.

**Comment by intrigus** [comment]
@badlogic The listeners are synchronized now, anything besides that?

**Comment by intrigus** [comment]
I resolved the merge conflict.

**Pull request merged**